industrial_ci/src/tests/source_tests.sh

```diff
@@ -79,8 +79,17 @@ rosdep --version
 if ! [ -d /etc/ros/rosdep/sources.list.d ]; then
     sudo rosdep init
 fi
-ret_rosdep=1
-rosdep update || while [ $ret_rosdep != 0 ]; do sleep 1; rosdep update && ret_rosdep=0 || echo "rosdep update failed"; done
+
+update_opts=()
+case "$ROS_DISTRO" in
+"hydro"|"jade")
+    if rosdep update --help | grep -q -- --include-eol-distros; then
```

Comment:
I found this `if` clause a thoughtful addition :).

In `else` situation for this `if`, I assume `rosdep` is old so all rosdep key still get fetched and no error occurs, thus we don't need `else` handling?